persistence/src/commonMain/db_user/com/wire/kalium/persistence/Conversations.sq

```diff
@@ -393,9 +393,15 @@ AND Conversation.mls_group_state IS 'ESTABLISHED'
 ORDER BY Conversation.type DESC LIMIT 1;
 
 getMLSGroupIdByUserId:
-SELECT Conversation.mls_group_id FROM Member
-JOIN Conversation ON Conversation.qualified_id = Member.conversation
-WHERE Conversation.mls_group_id IS NOT NULL AND mls_group_state IS 'ESTABLISHED' AND Member.user = :userId LIMIT 1;
+SELECT mls_group_id FROM
+	(SELECT COUNT(Member.user) AS users_amout, Conversation.mls_group_id, Conversation.mls_group_state FROM Member
```

Comment:
Two points:
1. Please let's make sure this query is called whenever it's necessary. Mls and e2ei enabled for example. 
2. Can we think of something better to detect if there is group that we're not part of it? E.g. a state somewhere in the db that we can filter based on it?

Reply:
1. this is already done in UseCases that are calling this query (`GetUserE2eiCertificatesUseCaseImpl`, `GetUserE2eiCertificateStatusUseCaseImpl`). 

2. I was looking for the best option here. We could make a query to `ConversationDetails` instead of `Conversation` table, and check there `self_role != null` but it's some workaround IMO. 
Or we could store somewhere conversations that user was removed from, BUT:
- this will add more complexity (we'll need to make sure this state is updated properly) and 
- creates additional source of truth, which might add more problems.


P.S. I agree this query is not the most beautiful query in the world, but we have a lot of queries and triggers that are much more complex ;P And this one does exactly what we need, without any additional changes and workarounds.

Reply:
Thanks for the clarification, I would go with this instead of fetching the ConversationDetails [ it's a much bigger boy]